src/citrine/resources/file_link.py

```diff
@@ -217,83 +226,23 @@ def _get_path_from_file_link(self, file_link: FileLink,
                                  *,
                                  action: str = None) -> str:
         """Build the platform path for taking an action with a particular file link."""
-        # Use this sessions project/dataset credentials and the URL's file / version
-        file_id, version_id = self._get_ids_from_url(file_link.url)
-        if file_id is None:
-            raise ValueError("FileLink did not contain a Citrine platform file URL.")
+        # Use this session's project/dataset credentials and the URLs file / version
+        if file_link.uid is not None:
+            file_id = file_link.uid
+            version_id = file_link.version
+        else:
+            file_id, version_id = self._get_ids_from_url(file_link.url)
+            if file_id is None:
+                raise ValueError("FileLink did not contain a Citrine platform file URL.")
         return self._get_path(uid=file_id, version=version_id, action=action)
 
     def build(self, data: dict) -> FileLink:
         """Build an instance of FileLink."""
+        # Use this chance to construct a URL from platform metadata
+        if 'url' not in data:
+            data['url'] = self._get_path(uid=data["id"], version=data["version"])
```

Comment:
A reasonable alternative approach here would be to have the collection builder insert a `project` and `dataset` field and then have the object builder use those to assemble a URL.  That probably matches responsibilities better, but would presumably have the FileLink access either `_get_path` or `_path_template`, which would technically be a private space violation.